server/src/index.ts

```diff
@@ -298,13 +315,66 @@ app.get("/api/searchPublicActivities", async (req: Request, res: Response) => {
   });
 });
 
-app.get("/api/loadPromotedContent", (req: Request, res: Response) => {
-  res.send({
-    success: true,
-    carouselData: {},
-  });
+app.get("/api/loadPromotedContent", async (req: Request, res: Response) => {
+  const loggedInUserId = Number(req.cookies.userId);
+  const isAdmin = loggedInUserId ? await getIsAdmin(loggedInUserId) : false;
+  const carouselData = await loadPromotedContent(isAdmin);
+  res.send(carouselData);
 });
 
+app.post("/api/addPromotedContent", async (req: Request, res: Response) => {
+  const groupId = req.body.groupId;
+  const activityId = req.body.activityId;
+  const { success, message } = await addPromotedContent(groupId, activityId);
```

Comment:
Throw errors in `addPromotedContent` and catch them here. (You can add more info to messages before throwing.) Here, catch and determine if it is an error you know about. If not, send it to `next()`.